bokeh/charts/data_source.py

```diff
@@ -167,9 +167,7 @@ def get_values(self, selection):
             :class:`pandas.DataFrame`
 
         """
-        if selection in special_columns:
-            return special_columns[selection](self.data)
-        elif isinstance(selection, str):
+        if isinstance(selection, str):
             return self.data[selection]
         elif isinstance(selection, list) and len(selection) == 1:
```

Comment:
So this code never worked. If selection was a list it would raise an exception `TypeError: unhashable type: list` on L170 (due to check the existence of a list in a dictionary)